src/main/js/table.ts

```diff
@@ -0,0 +1,37 @@
+/*
```

Comment:
Why is this file called table? I think this logic needs some unit tests.

Reply:
Naming.. It was time.ts at first and after some refactorings i kinda stuck at table. Suggestion?

Reply:
`time.ts` sounds good to me